jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1627,12 +1626,37 @@ protected Password getPassword(String realm)
      *
      * @param password the password string
      * @return the new Password object
+     * @deprecated use {#link newCredential} instead.
      */
+    @Deprecated(since = "12.0.13", forRemoval = true)
     public Password newPassword(String password)
     {
         return new Password(password);
     }
 
+    /**
+     * Returns the credential object for the given realm.
+     *
+     * @param realm the realm
+     * @return the Credential object
+     */
+    protected Credential getCredential(String realm)
+    {
+        String password = System.getProperty(realm);
+        return password == null ? null : newCredential(password);
+    }
+
+    /**
+     * Creates a new Credential object.
+     *
+     * @param password the password string
+     * @return the new Credential object
+     */
+    public Credential newCredential(String password)
```

Comment:
This should be `private`, as we now have a `ServiceLoader` mechanism to allow for customizations.